lightning/src/ln/reorg_tests.rs

```diff
@@ -0,0 +1,154 @@
+		let node_2_commitment_txn = nodes[2].node.channel_state.lock().unwrap().by_id.get_mut(&chan_2.2).unwrap().channel_monitor().get_latest_local_commitment_txn();
+		assert_eq!(node_2_commitment_txn.len(), 2); // 1 local commitment tx, 1 Received HTLC-Claim
+		assert_eq!(node_2_commitment_txn[0].output.len(), 2); // to-remote and Received HTLC (to-self is dust)
+		check_spends!(node_2_commitment_txn[0], chan_2.3);
+		check_spends!(node_2_commitment_txn[1], node_2_commitment_txn[0].clone());
+
+		// Give node 1 node 2's commitment transaction and get its response (timing the HTLC out)
+		nodes[1].block_notifier.block_connected(&Block { header, txdata: vec![node_2_commitment_txn[0].clone()] }, CHAN_CONFIRM_DEPTH + 1);
+		let node_1_commitment_txn = nodes[1].tx_broadcaster.txn_broadcasted.lock().unwrap();
+		assert_eq!(node_1_commitment_txn.len(), 3); // ChannelMonitor: 1 offered HTLC-Timeout, ChannelManger: 1 local commitment tx, 1 Offered HTLC-Timeout
+		assert_eq!(node_1_commitment_txn[1].output.len(), 2); // to-local and Offered HTLC (to-remote is dust)
+		check_spends!(node_1_commitment_txn[1], chan_2.3);
+		check_spends!(node_1_commitment_txn[2], node_1_commitment_txn[1].clone());
+		check_spends!(node_1_commitment_txn[0], node_2_commitment_txn[0]);
+
+		// Confirm node 2's commitment txn (and node 1's HTLC-Timeout) on node 1
+		nodes[1].block_notifier.block_connected(&Block { header, txdata: vec![node_2_commitment_txn[0].clone(), node_1_commitment_txn[0].clone()] }, CHAN_CONFIRM_DEPTH + 1);
+		// ...but return node 2's commitment tx (and claim) in case claim is set and we're preparing to reorg
+		node_2_commitment_txn
+	};
+	check_closed_broadcast!(nodes[1], false); // We should get a BroadcastChannelUpdate (and *only* a BroadcstChannelUpdate)
+	headers.push(header.clone());
+	// At CHAN_CONFIRM_DEPTH + 1 we have a confirmation count of 1, so CHAN_CONFIRM_DEPTH +
+	// ANTI_REORG_DELAY - 1 will give us a confirmation count of ANTI_REORG_DELAY - 1.
+	for i in CHAN_CONFIRM_DEPTH + 2..CHAN_CONFIRM_DEPTH + ANTI_REORG_DELAY - 1 {
+		header = BlockHeader { version: 0x20000000, prev_blockhash: header.bitcoin_hash(), merkle_root: Default::default(), time: 42, bits: 42, nonce: 42 };
+		nodes[1].block_notifier.block_connected_checked(&header, i, &vec![], &[0; 0]);
+		headers.push(header.clone());
+	}
+	check_added_monitors!(nodes[1], 0);
+	assert_eq!(nodes[1].node.get_and_clear_pending_events().len(), 0);
+
+	if claim {
+		// Now reorg back to CHAN_CONFIRM_DEPTH and confirm node 2's broadcasted transactions:
+		for (height, header) in (CHAN_CONFIRM_DEPTH + 1..CHAN_CONFIRM_DEPTH + ANTI_REORG_DELAY - 1).zip(headers.iter()).rev() {
+			nodes[1].block_notifier.block_disconnected(&header, height);
+		}
+
+		header = BlockHeader { version: 0x20000000, prev_blockhash: Default::default(), merkle_root: Default::default(), time: 42, bits: 42, nonce: 42 };
+		nodes[1].block_notifier.block_connected(&Block { header, txdata: claim_txn }, CHAN_CONFIRM_DEPTH + 1);
```

Comment:
Why no connect just one block here ? should be enough

Reply:
Make sure there's no off-by-one, not that it would matter a ton either way, but nice to test that nothing happens and then suddenly one more block later it does.

Reply:
We do only connect one block here? We just connect it at height CHAN_CONFIRM_DEPTH + 1.